test/testdata/java_archived_book_name.baseline

```diff
@@ -101,6 +104,22 @@ public class ArchivedBookName extends BookName {
     return PATH_TEMPLATE.matches(formattedString);
   }
 
+  public Map<String, String> getFieldValuesMap() {
+    if (fieldValuesMap != null) {
+      return fieldValuesMap;
+    }
+    ImmutableMap.Builder<String, String> fieldMapBuilder = ImmutableMap.builder();
```

Comment:
This probably doesn't matter much, but I think we should "double-check-lock" this so that only one map gets created. See example [here](https://github.com/google/auto/blob/master/value/src/main/java/com/google/auto/value/extension/memoized/Memoized.java#L69).

Currently two threads can both observe `fieldValuesMap` being null, then race to create the map.

Reply:
neat neat! implemented.